src/utils.jl

```diff
@@ -0,0 +1,73 @@
+"""
+    filterpeaks!(pks, mask) -> pks
```

Comment:
The reason for not documenting the min max method was because if someone wants to filter by min or max alone, they will do it when they compute the feature. It therefore has very little value in the public API. But I have now written that in the docstring, for both methods.

Reply:
We can document both, no problem.

In your proposed signatures you are using rather complicated syntax. This is not all all essential information, but _puts off newcomers_ (I am starting to sound like a broken record here, sorry about that). I think that something like the following is preferrable:
```
    filterpeaks!(pks min, max, feature) where {NT<:NamedTuple} -> pks::NamedTuple
    filterpeaks!(pks, mask) -> pks::NamedTuple
```

By the way, great idea to add the type annoration to the return argument. That helps clearing up the signature a lot, but because both are named `pks`, it still indicated the expected type of the input. Love it.